clean_air/util/file_converter.py

```diff
@@ -0,0 +1,106 @@
+"""This converts xcel data files to JSON file types for processing."""
+import pandas as pd
+import openpyxl
+import json
+import datetime
+import xarray as xr
+from json import JSONEncoder
+from collections import OrderedDict
+
+
+def read_excel_data(filepath):
+    """
+    Reads in data from excel spreadsheets and holds as temporary
+    pandas.DataFrame object.
+    """
+    temp_dataframe = pd.read_excel(filepath, engine='openpyxl')
+    print("reading today's excel data file and committing to memory...")
+    return pd.DataFrame(temp_dataframe)
+
+
+def read_netcdf_data(filepath):
+    """Reads in netcdf data from somewhere and holds in an xarray
+    something..."""
+    # TODO: Find details of ticket and complete the description of this function
+    temp_dataframe = xr.open_dataset(filepath)
+    print("reading today's netcdf data file and committing to memory...")
+    return xr.DataArray(temp_dataframe)
+
+
+def save_as_json(self):
+    """
+    Uses data held in pandas DataFrame to enter into form template and
+    save as JSON string.
+    """
+    # iterate through each row to split into separate dataframes and save in
+    # clean_air/data/json_data/:
+    for r, row in enumerate(temp_xl_df.iterrows()):
```

Comment:
What is `self` and what is `temp_x1_df`?

`self` is the usual name for "the object instance that this is a method of", but this isn't a method (we're not in a class block)

`temp_x1_df` is a free variable, not a local in this function. It's actually finding the variable of the same name defined at the bottom of this file, which is presumably only there for testing and wouldn't be there in the end.

In fact, because this function is called as `save_as_json(temp_x1_df)`, `self` and `temp_x1_df` are exactly the same object!

Reply:
'temp_xl_df' is a variable I accidentally pushed when I started work on #143 and then realised I'd missed some files for this PR.  This should be called 'temp_df' and is the temporary dataframe that is passed in after the datafile is read.  The 'xl' is in there now to highlight the difference between it being the excel datafile or the netcdf datafile.  I really need more information about what is supposed to be happening before and after these conversions so that I can structure the whole thing more concisely.

Using 'self' was the only way that I could actually construct the dictionary in order to actually write to a json file.  I understand that it's not good python structure and doesn't really make much sense, but it does work.  In my defence, firstly as more converters need to be written I expect this file to develop towards a class structure anyway, but secondly, I used this 'self' trick when I was trying to write the outfile using pandas still and nothing was working, so I'll see if i can use a less confusing (and more pythonic) method when writing out using the json handler.

And yes, I did get quite confused when writing and changing all the variable names, so I'll try and sort this all out.  It is messy, you are quite right, and I don't like things to be messy.

Reply:
I wasn't concerned about the name of the variables so much as the fact that it was (and still is according to the latest commit pushed) using a variable from the wrong scope.
Essentially the difference between
```python
def save_as_json(some_data):
    for r, row in some_data.iterrows():
        ...
```
instead of
```python
def save_as_json():
    for r, row in some_data.iterrows():
        ...

some_data = ...
save_as_json()
```

Reply:
Sorry, you'll have to bear with me here, still trying to find my feet with python again.  The top option here looks correct to me, is that right?

Reply:
Although I am still somewhat confused, mainly with the flow of the operations.  The structure of the file so far is working towards an automated script which uses save_as_json(), but that function can also be accessed by manually passing in some data, under the assumption that we may use this again somewhere else.

I think the end-goal of this function is important.  If it is only ever going to be used in this script for this specific operation, then I can hone it a bit so it's more direct and more restrictive, but at the moment I'm trying to leave it a bit open and flexible so that we can use it later if we need to.  I'm not really sure where to go from here...

Reply:
Yep, it's only making use of what was passed in or what was defined inside the function.

The second one uses a "free variable", which only works if a variable with that same name is defined outside the function. Which in practice it wouldn't - the module would just define a bunch of functions and classes, it wouldn't call any functions or load any data. Currently it does, that's why I assume that was just for convenient testing.

Reply:
I'm a little concerned that I'm not doing pushes right and I keep making changes you asked for and then sending you something different.  I just want to have a look at the latest commit to make sure it's what I meant to send you...

Reply:
I've had a look at the changes I last pushed up, and they are what I was expecting to push, so that's good.  However, I am still not exactly sure what is wrong with this function now.  The whole 'self' thing was weird and I got rid of it and used a normal variable instead, and the definition of the function matches the structure you showed an example of (i.e. no free variable, you have to define the data you are sending to the function in the call signature).

What exactly is wrong with it atm?